package.json

```diff
@@ -9,11 +9,12 @@
   "scripts": {
     "start": "cd examples && parcel serve ./*.html --dist-dir ./dev-bundle/ --no-cache --no-hmr",
     "build-examples": "cd examples && parcel build ./*.html --dist-dir ./bundle/ --public-url . --no-cache --no-content-hash",
-    "build": "rollup -c",
+    "build": "rollup -c && npm run types",
     "lint": "eslint \"./src/**/*.js\" \"./examples/*.js\"",
     "benchmark": "node ./benchmark/benchmark.js",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "prepublishOnly": "npm run build"
+    "prepublishOnly": "npm run build",
+    "types": "npx -p typescript tsc -d"
```

Comment:
What is this doing? If d.ts files are not written with actual types what is the value?

Reply:
Run it and you will see `index.module.d.ts` in a build folder. It will contain all the type definition for a `index.module.js` file. Having both js and d.ts files user can import it as if it is just a typescript file

Reply:
I'm very familiar with d.ts files but I've run the build and the d.ts file generated is not precise, exposes a lot of "private" variables, and uses `any` excessively and incorrectly. If this is what tsc can derive from just looking at a js file there shouldn't be a lot of value to building it out ahead of time.

If we'd like to add typing to the project I'd prefer to have a hand-maintained index.d.ts file that exposes just the public, usable parts of the AP. Similar to how this project is organized:

https://github.com/gkjohnson/urdf-loaders/tree/master/javascript/src

Reply:
For sure hand-maintained file will be better, but it requires a strong commitment. This solution is just a first step and people already can use the lib with typescript even if types are not perfect

Reply:
The public API that will need types is very thin. If types are going to be added to the project I'd prefer to do it in the right way.